analyzers/src/SonarAnalyzer.CSharp/Rules/ParameterValidationInMethodWalker.cs

```diff
@@ -44,36 +44,39 @@ internal class ParameterValidationInMethodWalker : SafeCSharpSyntaxWalker
 
         public IEnumerable<Location> ArgumentExceptionLocations => argumentExceptionLocations;
 
-        public ParameterValidationInMethodWalker(SemanticModel semanticModel)
-        {
-            this.semanticModel = semanticModel;
-        }
+        public ParameterValidationInMethodWalker(SemanticModel semanticModel) => this.semanticModel = semanticModel;
 
         public override void Visit(SyntaxNode node)
         {
-            if (this.keepWalking &&
-                !node.IsAnyKind(SubMethodEquivalents))  // Don't explore deeper if this node is equivalent to a method declaration
+            if (keepWalking
+                && !node.IsAnyKind(SubMethodEquivalents))  // Don't explore deeper if this node is equivalent to a method declaration
             {
                 base.Visit(node);
             }
         }
 
-        public override void VisitAwaitExpression(AwaitExpressionSyntax node)
-        {
-            this.keepWalking = false;
-        }
+        public override void VisitAwaitExpression(AwaitExpressionSyntax node) => keepWalking = false;
 
         public override void VisitThrowStatement(ThrowStatementSyntax node)
```

Comment:
What about *throw expressions*? Are these handled too?
```cs
_ = o ?? throw new ArgumentNullException();
```
(Hint: throw expressions were introduced in C# 7 (if I'm not mistaken).)

Reply:
good catch! https://github.com/SonarSource/sonar-dotnet/issues/6369

(I've labeled it C# 8 because it's the closest and I don't want to add a C# 7 label)